src/main/webapp/app/shared/participant-scores/participant-scores-distribution/participant-scores-distribution.component.ts

```diff
@@ -0,0 +1,318 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
+import { Color, ScaleType } from '@swimlane/ngx-charts';
+import { NgxChartsSingleSeriesDataEntry } from 'app/shared/chart/ngx-charts-datatypes';
+import { GradeType, GradingScale } from 'app/entities/grading-scale.model';
+import { GradingSystemService } from 'app/grading-system/grading-system.service';
+import { TranslateService } from '@ngx-translate/core';
+import { GradeStep } from 'app/entities/grade-step.model';
+import { GraphColors } from 'app/entities/statistics.model';
+
+@Component({
+    selector: 'jhi-participant-scores-distribution',
+    templateUrl: './participant-scores-distribution.component.html',
+    styleUrls: ['./participant-score-distribution.component.scss', '../../chart/vertical-bar-chart.scss'],
+})
+export class ParticipantScoresDistributionComponent implements OnInit, OnChanges {
+    @Input()
+    scores: number[];
+
+    @Input()
+    gradingScale?: GradingScale;
+
+    @Input()
+    isCourseScore = true;
+
+    @Input()
+    dataLabelFormatting?: (submissionCount: number) => string;
+
+    @Input()
+    onSelect?: (event?: any) => any;
+
+    @Input()
+    scoreToHighlight?: number;
+
+    @Output()
+    scoreHighlightingVisible: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    gradingScaleExists = false;
+    isBonus?: boolean;
+    ngxData: NgxChartsSingleSeriesDataEntry[];
+    yScaleMax: number;
+    height = 500;
+
+    yAxisLabel = this.translateService.instant('artemisApp.examScores.yAxes');
+    xAxisLabel = this.translateService.instant('artemisApp.examScores.xAxes');
+
+    helpIconTooltip: string;
+
+    readonly binWidth = 5;
+
+    ngxColor = {
+        name: 'Participation scores distribution',
+        selectable: true,
+        group: ScaleType.Ordinal,
+        domain: [],
+    } as Color;
+    backupDomain: string[];
+
+    constructor(private gradingSystemService: GradingSystemService, private translateService: TranslateService) {}
+
+    ngOnInit() {
+        this.translateService.onLangChange.subscribe(() => {
+            this.setupAxisLabels();
+        });
+    }
+
+    ngOnChanges() {
+        if (this.gradingScale) {
+            this.gradingScaleExists = true;
+            this.isBonus = this.gradingScale.gradeType === GradeType.BONUS;
+        }
+        if (!this.isCourseScore) {
+            this.height = 400;
```

Comment:
Why is the height different in this case?

Reply:
This is the height of the chart on the exam scores page and I wanted to keep it. In case of the course scores page, I found it to be a bit small. But this may be some avoidable complexity in the code.

Reply:
Can you please add a comment explaining why it is different so that other people in the future reading the code know about that?

Reply:
Sure, done in 40b5f67